File: pset6/sentimental-readability/readability.py
def main():
    # get user input
    inpt = input("Text: ")

    # call functions that count letters, words, and sentences
    letter_count = count_letters(inpt)
    word_count = count_words(inpt)
    sentence_count = count_sentences(inpt)

    # impliment Coleman-Liau index formula
    index = (0.0588 * (letter_count / word_count * 100) - 0.296 *
             (sentence_count / word_count * 100) - 15.8)

    # format index into printable format
    new_index = round(index)

    # print out result
    # print results greater than grade 16
    if (new_index >= 16):
        print("Grade 16+")
        return 0

    # print results less than grade 1
    if (new_index < 1):
        print("Before Grade 1")
        return 0

    # print all other grades in between
    else:
        print(f"Grade {new_index}")
        return 0


# function to count letters in text input
def count_letters(text):
    letter_count = 0
    i = 0
    while (len(text) > i):
        if (text[i].isalpha()):
            letter_count += 1
        i += 1

    return letter_count

# function to count words in text input


def count_words(text):
    word_count = 0
    i = 0
    while (len(text) > i):
        if (text[i] == ' '):
            word_count += 1
        i += 1

    word_count += 1
    return word_count

# function to count sentences in text input


def count_sentences(text):

    sentence_count = 0
    i = 0
    while (len(text) > i):
        if (text[i] == '.' or text[i] == '?' or text[i] == '!'):
            sentence_count += 1
        i += 1

    return sentence_count


if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>


// function prototypes
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

// main function
int main(void)
{
    // get user input
    string input = get_string("Text: ");

    //call functions that count letters, words, and sentences
    int letter_count = count_letters(input);
    int word_count = count_words(input);
    int sentence_count = count_sentences(input);

    // impliment Coleman-Liau index formula
    double index = 0.0588 * ((float)letter_count / (float) word_count * 100) - 0.296 *
    ((float)sentence_count / (float) word_count * 100) - 15.8;

    // format index into printable format
    int new_index = (int) round(index);

    // print out result
        // print results greater than grade 16
    if (new_index >= 16)
    {
        printf("Grade 16+\n");
        return 0;
    }
        // print results less than grade 1
    if (new_index < 1)
    {
        printf("Before Grade 1\n");
        return 0;
    }
        // print all other grades in between
    else
    {
        printf("Grade %i\n", new_index);
        return 0;
    }
}

// function to count letters in text input
int count_letters(string text)
{
    int letter_count = 0;
    int i = 0;
    while (strlen(text) > i)
    {
        if (isalpha(text[i]))
        {
            letter_count++;
        }
        i++;
    }
    return letter_count;
}
// function to count words in text input
int count_words(string text)
{
    int word_count = 0;
    int i = 0;
    while (strlen(text) > i)
    {
        if (text[i] == ' ')
        {
            word_count++;
        }
        i++;
    }
    word_count++;
    return word_count;
}
// function to count sentences in text input
int count_sentences(string text)
{
    int sentence_count = 0;
    int i = 0;
    while (strlen(text) > i)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentence_count++;
        }
        i++;
    }
    return sentence_count;
}